test/Billing.Test/Controllers/FreshdeskControllerTests.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System.Text.Json;
+using Bit.Billing.Controllers;
+using Bit.Core.Entities;
+using Bit.Core.Repositories;
+using Bit.Core.Settings;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using NSubstitute;
+using Xunit;
+
+namespace Bit.Billing.Test.Controllers
+{
+    public class FreshdeskControllerTests
+    {
+        private const string ApiKey = "TESTFRESHDESKAPIKEY";
+        private const string WebhookKey = "TEST_FRESHDESK_WEBHOOKKEY";
+        private const string TicketId = "TEST_FRESHDESK_TICKETID";
+        private const string Email = "TEST@EMAIL.COM";
+
+        private static (FreshdeskController, IUserRepository, IOrganizationRepository) CreateSut(
```

Comment:
It's possible that the `SutProvider` class will just generate the exact sut you want and you can override a couple of returns to get this same thing.

Check out `OrganizationUsersControllerTests` for some examples on the clean tests you can make.